### DISCUSSION TITLE
Cloudstack 4.20, kvm install problem.

### DISCUSSION THREAD

**Discussion by rongage1** [apache/cloudstack#10090] [Q&A] [answered]
I am trying to freshly install 4.20 host agent for KVM on an Oracle Linux host running R9U5.  It is not going well, specifically the networking setup is failing.  

The juxt of the problem is that the setup is trying to restart the network stack and using the entirely wrong command.  On Oracle, the proper way to restart the network stack is with "nmcli networking off && nmcli networking on".  There is no "network.service" unit defined on Oracle Linux (or on Centos Stream 9).  

With the setup failing at this step (restarting the network stack), the remainder of the host setup of course fails.

The data in /var/log/cloudstack/agent/setup.log:
DEBUG:root:execute:hostname -f
DEBUG:root:execute:awk '/MemTotal/ { printf "%.3f \n", $2/1024 }' /proc/meminfo
DEBUG:root:execute:ip a | grep "^\w" | grep -iv "^lo" | wc -l
DEBUG:root:execute:selinuxenabled
DEBUG:root:Failed to execute:
DEBUG:root:Management is not a network device, is it down?
DEBUG:root:execute:route -n|awk '/^0.0.0.0/ {print $2,$8}'
DEBUG:root:execute:ifconfig cloudbr0
DEBUG:root:Found default network device:cloudbr0
DEBUG:root:execute:ifconfig enp0s31f6
DEBUG:root:execute:setenforce 1
DEBUG:root:Failed to execute:setenforce: SELinux is disabled
DEBUG:root:execute:service network restart
DEBUG:root:Failed to execute:Redirecting to /bin/systemctl restart network.service
Failed to restart network.service: Unit network.service not found.

I welcome any suggestions to get past this hurdle.

Thanks!

Ron Gage
Westland, Michigan

**Comment by DaanHoogland** [comment]
wow, that is a bad find @rongage1 . As a workaround you could implement a simple service file that does `nmcli networking off && nmcli networking on` in the `restart` method. As far as I know it is implementing the `ExecReload` method, but do not take my word! 

limitation of warranty: this is a hack at best. A more permanent solution must be found.

**Reply by weizhouapache** [reply, marked as answer]
> wow, that is a bad find @rongage1 . As a workaround you could implement a simple service file that does `nmcli networking off && nmcli networking on` in the `restart` method. As far as I know it is implementing the `ExecReload` method, but do not take my word!
> 
> limitation of warranty: this is a hack at best. A more permanent solution must be found.

the file to setup cloudstack-agent networking is out of date
https://github.com/apache/cloudstack/blob/main/python/lib/cloudutils/serviceConfig.py

I suggest to configure linux bridges following the instruction on cloudstack documentation.
https://docs.cloudstack.apache.org/en/latest/quickinstallationguide/qig.html#configuring-the-network
